app/src/main/java/ml/docilealligator/infinityforreddit/utils/Utils.java

```diff
@@ -88,64 +90,116 @@ public static String modifyMarkdown(String markdown) {
+        final boolean hasCodeRanges = codeRanges != null && codeRanges.size() > 0;
+
         StringBuilder regexed = new StringBuilder(regexedMarkdown);
         boolean hasBracket = false;
         int nCarets = 0;
         int newLines = 0;
+        int offset = 0;
+        int escapes = 0;
+
+        outerLoop:
         for (int i = 0; i < regexed.length(); i++) {
             char currentChar = regexed.charAt(i);
-            if (hasBracket && currentChar == '\n') {
+
+            if (hasCodeRanges && currentChar == '^') {
+                for (var codeRange : codeRanges) {
+                    if (codeRange.start + offset <= i && i <= codeRange.end + offset) {
+                        continue outerLoop;
+                    }
+                }
+            }
+
+            if (currentChar == '\\') {
+                escapes++;
+                newLines = 0;
+            } else if (hasBracket && currentChar == '\n') {
                 newLines++;
                 if (newLines > 1) {
                     hasBracket = false;
                     nCarets = 0;
                     newLines = 0;
                 }
-            } else if (currentChar == '^') {
-                if (!(i > 0 && regexed.charAt(i - 1) == '\\')) {
-                    if (nCarets == 0 && i < regexed.length() - 1 && regexed.charAt(i + 1) == '(') {
-                        regexed.replace(i, i + 2, "<sup>");
-                        hasBracket = true;
-                    } else {
-                        regexed.replace(i, i + 1, "<sup>");
-                    }
-                    nCarets++;
+            } else if (currentChar == '^' && escapes % 2 == 0) {
+                if (nCarets == 0 && i < regexed.length() - 1 && regexed.charAt(i + 1) == '(') {
+                    regexed.replace(i, i + 2, "<sup>");
+                    hasBracket = true;
+                    i += 3;
+                    offset += 3;
+                } else {
+                    regexed.replace(i, i + 1, "<sup>");
+                    i += 4;
+                    offset += 4;
                 }
+                nCarets++;
             } else if (hasBracket && currentChar == ')') {
-                if (i > 0 && regexed.charAt(i - 1) == '\\') {
+                if (escapes % 2 != 0) {
                     hasBracket = false;
                     nCarets--;
                     continue;
                 }
                 hasBracket = false;
                 regexed.replace(i, i + 1, "</sup>");
                 nCarets--;
+                i += 5;
+                offset += 5;
+            } else if (hasCodeRanges && nCarets > 0 && exists(codeRanges, offset, i)) {
+                for (int j = 0; j < nCarets; j++) {
+                    regexed.insert(i, "</sup>");
+                    i += 6;
+                    offset += 6;
+                }
+                nCarets = 0;
```

Comment:
This doesn't work for ``a^(one `two` three)``. `three` should be superscripted.

Reply:
Thanks!

Reply:
I did a bit more testing:
``a^one`two`three`` - three should be superscripted
``a^one`two three`four`` - four is superscripted only on new Reddit. On old Reddit space stops the superscript and there is no code block

Reply:
Thank you, will look into these.